source/developer/interactive-message-buttons.rst

```diff
@@ -1,209 +1,4 @@
-  
-  1. **Identifying which action was triggered**. For example, a GitHub integration might store something like this in the context:
-
-    .. code-block:: text
-
-      {
-      "user_id": "rd49ehbqyjytddasoownkuqrxe",
-      "post_id": "gqrnh3675jfxzftnjyjfe4udeh",
-      "channel_id": "j6j53p28k6urx15fpcgsr20psq",
-      "team_id": "5xxzt146eax4tul69409opqjlf",
-      "context": {
-        "repo": "mattermost/mattermost-server"
-        "pr": 1234,
-        "action": "merge"
-        }
-      }
-      
-  When the message button is clicked, your integration sends a request to the specified URL with the intention to merge the pull request identified by the context.
-
-  2. **Authenticating the server**. An important property of the context parameter is that it's kept confidential. Hence, if your integration is not behind a firewall, you could add a token to your context without users ever being able to see it:
-
-    .. code-block:: text
-
-      {
-      "user_id": "rd49ehbqyjytddasoownkuqrxe",
-      "post_id": "gqrnh3675jfxzftnjyjfe4udeh",
-      "channel_id": "j6j53p28k6urx15fpcgsr20psq",
-      "team_id": "5xxzt146eax4tul69409opqjlf",
-      "context": {
-        "repo": "mattermost/mattermost-server"
-        "pr": 1234,
-        "action": "merge",
-        "token": "somerandomlygeneratedsecret"
-        }
-      }
-   
-  Then, when your integration receives the request, it can verify that the token matches one that you previously generated and know that the request is legitimately coming from the Mattermost server and not forged.
-
-  Depending on the application, integrations can also perform authentication statelessly with cryptographic signatures such as:
-
-    .. code-block:: text
-
-      {
-      "user_id": "rd49ehbqyjytddasoownkuqrxe",
-      "post_id": "gqrnh3675jfxzftnjyjfe4udeh",
-      "channel_id": "j6j53p28k6urx15fpcgsr20psq",
-      "team_id": "5xxzt146eax4tul69409opqjlf",
-      "context": {
-        "repo": "mattermost/mattermost-server"
-        "pr": 1234,
-        "action": "merge",
-        "signature": "mycryptographicsignature"
-        }
-      }
-
-  It's also possible for integrations to do both of these things with a single token and use something like this as context:
-
-    .. code-block:: text
-
-      {
-      "user_id": "rd49ehbqyjytddasoownkuqrxe",
-      "post_id": "gqrnh3675jfxzftnjyjfe4udeh",
-      "context": {
-        "action_id": "someunguessableactionid"
-        }
-      }
-
-  Then, when the integration receives the request, it can act based on the action id.
-
-Tips and Best Practices
-------------------------
-
-1. The external application may be written in any programming language. It needs to provide a URL which receives the request sent by your Mattermost server and responds with in the required JSON format.
-2. Message attachments allow for rich formatting, but don't go overboard. Use the least amount of formatting and number of action buttons required for your integration post.
```

Comment:
Consider revising this statement to provide more of a reason why. ie: "Use the least amount of number of action buttons required and clean formatting to ensure your users can utilize the action buttons without confusion."